spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -20,33 +20,141 @@
+An application using Reactive Messaging is composed by CDI beans consuming, producing and processing _messages_.
 
-* Both are designed for a technology landscape where messaging was a heavy weight process used on the edge of system to hand control of a transaction from system to another, consequently there is a lot of boilerplate associated with using these technologies.
-* It is assumed their design that consistency is handled using distributed transactions. However, modern message brokers, popular in microservice deployments, such as Apache Kafka, Amazon Kinesis and Azure Event Hubs, do not support XA transactions, rather, message acknowledgement is handled using offsets with at least once delivery guarantees.
-* They do not have support for asynchronous IO, it is assumed that message processing is done on a single thread, however many modern specs are moving to asynchronous IO.
+These messages can be _internal_ messages and be received/forwared from/to different brokers.
 
-Hence a lighter weight, reactive solution to messaging is desirable for MicroProfile to ensure microservices written using MicroProfile are able to meet the demands required by the architecture.
+image::overall.png[Overall architecture]
 
-=== Design
+Application's beans contain methods annotated with `@Incoming` and `@Outgoing` annotations. 
+A method with an `@Incoming` annotation consumes messages from a _channel_. 
+A method with an `@Outgoing` annotation produces messages, it emits messages to a _channel_.
+A method with both an `@Incoming` and an `@Outgoing` annotation is a message processor, it consumes messages from a _channel_, does some transformation to them, and emits messages to another_channel_.
+
+==== Channel
+
+A _channel_ is a name indicating from expected messages are consumed or to where messages are sent.
+Channels are opaque `Strings`.
+
+There are two type of channel:
+
+* Internal channels are local to the application. 
+It allows implementing multi-step processing where several _beans_ from the same application form a chain of processing.
+* Channels can be _connected_ to remote brokers or various message transport layer such as Apache Kafka or to an AMQP broker.
+These channels are managed by _connectors_.
+
+==== Message
+
+At the core of the Reactive Messaging specification is the concept of _message_. 
+A _message_  is an enveloppe wrapping a _payload_. 
+A message is sent to a specific channel and, when received and processed successfuly, _acknowledged_.
+
+Reactive Messaging application components are addressable recipients await the arrival of messages on a channel and react to them, otherwise lying dormant. 
+
+Message are represented by the `org.eclipse.microprofile.reactive.messaging.Message` class. 
+This interface is kept minimal on purpose aiming _connectors_ to provide their own with additional metadata.
```

Comment:
This interface is intentionally kept minimal. The aim is that _connectors_ will provide their own implementations with additional metadata that is relevant to that connector.